### PR TITLE
Remove redundant serviceAccountName assignment

### PR DESCRIPTION
**Pull request by hougangliu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/969)

### CODE REVIEW

File: pkg/webhook/v1alpha3/pod/inject_webhook.go [hunk 1]
```diff
@@ -143,7 +143,6 @@ func (s *sidecarInjector) Mutate(pod *v1.Pod, namespace string) (*v1.Pod, error)
 	}
 	mutatedPod.Spec.Containers = append(mutatedPod.Spec.Containers, *injectContainer)
 
-	mutatedPod.Spec.ServiceAccountName = pod.Spec.ServiceAccountName
```
Comment (hougangliu) [inline pkg/webhook/v1alpha3/pod/inject_webhook.go]: I had questioned it https://github.com/kubeflow/katib/pull/716#discussion_r317462741, but never go further. By test, it proves this line can be deleted

**Comment by johnugeorge** [comment]
/lgtm
I don't see the need either. Not sure why @wuchunghsuan added this

**Pull request merged**